urllib3/contrib/pyopenssl.py

```diff
@@ -167,13 +178,60 @@ def __init__(self, connection, socket, suppress_ragged_eofs=True):
         self.socket = socket
         self.suppress_ragged_eofs = suppress_ragged_eofs
         self._makefile_refs = 0
+        self._closed = False
 
     def fileno(self):
         return self.socket.fileno()
 
-    def makefile(self, mode, bufsize=-1):
-        self._makefile_refs += 1
-        return _fileobject(self, mode, bufsize, close=True)
+    if _fileobject:  # Platform-specific: Python 2
+        def makefile(self, mode, bufsize=-1):
+            self._makefile_refs += 1
+            return _fileobject(self, mode, bufsize, close=True)
+    else:  # Platform-specific: Python 3
```

Comment:
Sure, that's fine (though I think the `as` syntax is incorrect in your example, should be `import foo as bar, baz as quux` I think?).

Also, is it worth including _decref_socketios? that seems like a small enough standalone thing to keep inline.